src/Nox.Solution/NoxSolution.cs

```diff
@@ -39,6 +39,53 @@ internal void Validate()
         return fullRelationshipModels;
     }
 
+    public List<EntityRelationshipWithType> GetOwnedRelationshipsToCreate(
+        Func<string, Type?> getTypeByNameFunc,
```

Comment:
now looking at this, it seems to be an obscured thin

lets just rename to getEntityTypeByName we need to think how to improve this